src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -0,0 +1,646 @@
+    logger.debug("Checking for mismatched utility ids")
+    assert ~utils_eia860.duplicated(subset=["report_date", "utility_id_eia"]).any()
+    # Make a dictionary of PUDL id to EIA id (in some cases it is 1:m)
+    eia_id_list_series = utils_eia860.groupby("utility_id_pudl").apply(
+        lambda x: x.utility_id_eia.unique().tolist()
+    )
+    eia_id_dict = dict(zip(eia_id_list_series.index, eia_id_list_series))
+    # Add a column that extracts the utility id eia from record_id_eia_override_1. This
+    # gets a little wonky because sometimes it ends with the eia id: _55555 and
+    # sometimes it ends with a string: _55555_retired. This regex gets both and
+    # converts them to integers.
+    only_overrides[
+        "utility_id_eia_override"
+    ] = only_overrides.record_id_eia_override_1.str.extract(r"(\d+_*[a-z]*$)")[
+        0
+    ].str.extract(
+        r"(\d+)"
+    )
+    # For some reason I can't just add .astype("Int64 to the prior step...")
+    only_overrides[
+        "utility_id_eia_override"
+    ] = only_overrides.utility_id_eia_override.astype("Int64")
+
+    assert (
+        len(
+            mismatched_utilities := only_overrides[
+                ~only_overrides.apply(
+                    lambda x: x.utility_id_eia_override
+                    in eia_id_dict[x.utility_id_pudl],
+                    axis=1,
+                )
+            ]
+        )
+        == 0
+    ), f"Found mismatched utilities: \
+    {mismatched_utilities.record_id_eia_override_1}"
+
+    # Make sure the year in the EIA id overrides match the year in the report_year
+    # column.
+    logger.debug("Checking that year in override id matches report year")
+    year_ser = only_overrides.record_id_eia_override_1.str.extract(r"(_20\d{2})")[
+        0
+    ].str.replace("_", "")
+    year_ser_int = pd.to_numeric(year_ser, errors="coerce").astype("Int64")
+    assert (
+        len(bad_eia_year := only_overrides["report_year"].compare(year_ser_int)) == 0
+    ), f"Found record_id_eia_override_1 values that don't correspond to the right \
+        report year:\
+        {[only_overrides.iloc[x].record_id_eia_override_1 for x in bad_eia_year.index]}"
+
+    # If you don't expect to override values that have already been overridden, make
+    # sure the ids you fixed aren't already in the training data.
+    if not expect_override_overrides:
+        existing_training_eia_ids = training_data.record_id_eia.dropna().unique()
+        _check_id_consistency(
+            "eia", only_overrides, existing_training_eia_ids, "already in training"
+        )
+        existing_training_ferc_ids = training_data.record_id_ferc1.dropna().unique()
+        _check_id_consistency(
+            "ferc", only_overrides, existing_training_ferc_ids, "already in training"
+        )
+
+    # Only return the results that have been verified
+    verified_connections = validated_connections[
+        validated_connections["verified"]
+    ].copy()
```

Comment:
So I thought about where to put this, and I ended up putting it at the end because I wanted a user to be able to test record matches they had not yet verified. The verified ones are the only ones that will get integrated into the training data, but maybe you want to test some non verified matches too. There are a bunch of ways to do this, but I'm going to leave it as is for now.